### ISSUE TITLE
Rule Change: Add frontmatterTitle option to heading-increment

### ISSUE THREAD

**Issue by xbinaryx** [eslint/markdown#443]
### What rule do you want to change?

heading-increment

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```markdown
---
title: "level 1"
---

### level 3
```

### What does the rule currently do for this code?

No error is reported.

### What will the rule do after it's changed?

The rule will treat the front matter title as a level 1 heading, so it will report an error for skipping heading levels (jumping from 1 to 3).

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

See [MD001](https://github.com/DavidAnson/markdownlint/blob/next/doc/md001.md)

_No response_

**Comment by lumirlumir** [comment]
Sounds good to me 👍

Would like to hear a few more opinions from @eslint/eslint-team before moving forward.

**Comment by nzakas** [comment]
Sounds good to me. :+1:
